# Project 6 -- Kennedy Zapalac

**TA Help:** N/A
    
**Collaboration:** N/A

## Question 1
A. How many columns and rows are in the following dataset: `/anvil/projects/tdm/data/restaurant/orders.csv`.

In [4]:
%%bash
# Question 1
# print the number of fields for each row, then count number of unique fields, which should output the number of rows
# if each row has the same number of fields, which this file does
# then print the first field (# of rows) and the second field (# of columns)
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{print NF}' | uniq -c | awk -F" " '{print "rows: "$1; print "columns: "$2}'
# alternative if you're sure that each row has the same # of fields
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{ } END{print "rows: "NR; print "columns: "NF}'

rows: 135304
columns: 26


There are 135,304 rows and 26 columns in this dataset.

## Question 2
A. Please list all possible values of "Location Type" in the file `/anvil/projects/tdm/data/restaurant/orders.csv`

In [30]:
%%bash
# Question 2
# print the 25th field, the location type
# remove the header
# sort and count unique entries
# print the first 2 lines
# print the unique location types and number of orders for each location
# format using printf, giving the first output 17 spaces and the second output 16 spaces from the left (- indicates left alignment)
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{print $25}' | grep -v LOCATION_TYPE | sort | uniq -c | 
awk 'BEGIN{print "Location Type    Number of Orders"; print "-------------    ----------------"} 
    {printf "%-17s%-16s\n", $2, $1}'

Location Type    Number of Orders
-------------    ----------------
                 48893           
Home             51484           
Other            16669           
Work             18257           


It looks like there are a lot of blank entries for the location type. Home is the most frequent location type for restaurant orders.

## Question 3
A. What is the year range for the data in the dataset: `/anvil/projects/tdm/data/restaurant/orders.csv`

In [30]:
%%bash
# Question 3
# grab the 21st field which contains the delivery date
# remove the header
# print just the date without the time
# grab just the year
# sort the years
# output the unique years
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{print $21}' | grep -v delivery_date | awk -F" " '{print $1}' | awk -F"-" '{print $1}' | sort | uniq


2019


All the data in this dataset comes from the year 2019.

## Question 4
A. What is the sum of the order amounts for each year in the data set `/anvil/projects/tdm/data/restaurant/orders.csv`

In [29]:
%%bash
# Question 4
# the only year in this dataset is 2019, so I just have to sum the grand total (field 4) for every row and format the output
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{print "Year Summary of Orders in dollars"} {total += $4} END{print "2019 $" total}'

Year Summary of Orders in dollars
2019 $2.08022e+06


The sum of the order amounts for 2019 (the only year in the dataset) is $2,080,220.

## Question 5
Please extract both the years and months for the file: `/anvil/projects/tdm/data/whin/observations.csv`

In [53]:
%%bash
# Question 5
# pipe the output of cat
# grab the 3rd field, which contains the observation date and time
# print just the year and month
# sort year and month
# count unique combinations of year and month
# format print output
cat /anvil/projects/tdm/data/whin/observations.csv | awk -F"|" '{print $3}' | grep -v observation_time | awk -F"-" '{print $1, $2}' | sort | uniq -c | awk 'BEGIN{print "Month and Year    Number of Occurrences"; print "--------------    ---------------------"} {printf "%-4s-%-13s%-21s\n", $2, $3, $1}'

Month and Year    Number of Occurrences
--------------    ---------------------
2019-07           2002                 
2019-08           3969                 
2019-09           5749                 
2019-10           5948                 
2019-11           5731                 
2019-12           5918                 
2020-01           5482                 
2020-02           5008                 
2020-03           5436                 
2020-04           19326                
2020-05           42819                
2020-06           45557                
2020-07           50557                
2020-08           53288                
2020-09           53400                
2020-10           58283                
2020-11           57501                
2020-12           59509                
2021-01           61254                
2021-02           60306                
2021-03           81536                
2021-04           80116                
2021-05           83307                


It seems like 2021-05 is the most frequently occurring year and month combination in this dataset

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.